components/match2/wikis/ageofempires/match_group_input_custom.lua

```diff
@@ -12,520 +12,342 @@ local Game = require('Module:Game')
+		local manualPlayersInput = MatchGroupInputUtil.extractManualPlayersInput(match, opponentIndex, opponentInput)
+		substitutions = manualPlayersInput.substitutions
+		-- Change compared to commons MatchGroupInputUtil.readOpponent
+		local template = mw.ext.TeamTemplate.raw(opponent.template or '') or {}
+		opponent.players = MatchGroupInputUtil.readPlayersOfTeamNew(
+			template.page or '',
+			manualPlayersInput,
+			options,
+			{timestamp = match.timestamp, timezoneOffset = match.timezoneOffset}
+		)
 	end
 
-	local scores = Array.mapIndexes(function(opponentIndex)
-		local opponent = match['opponent' .. opponentIndex]
-		if not opponent then
-			return nil
-		end
-		return match['opponent' .. opponentIndex].score or -1
+	if options.pagifyPlayerNames then
+		Array.forEach(opponent.players or {}, function(player)
+			player.pageName = Page.pageifyLink(player.pageName)
+		end)
 	end
-	)
-	local maxScore = math.max(unpack(scores) or 0)
-	-- if we have a positive score and the match is finished we also have a winner
-	if maxScore > 0 then
-		local maxIndexFound = false
-		for opponentIndex, score in pairs(scores) do
-			if maxIndexFound and score == maxScore then
-				match.winner = 0
-				break
-			elseif score == maxScore then
-				maxIndexFound = true
-				match.winner = opponentIndex
-			end
-		end
+
+	local record = MatchGroupInputUtil.mergeRecordWithOpponent(opponentInput, opponent, substitutions)
+
+	-- no need to pagify non opponent names as for literals it is irrelevant
+	-- and for party opponents it comes down to pagifying player names
+	if options.pagifyTeamNames and opponent.type == Opponent.team then
+		record.name = Page.pageifyLink(record.name)
 	end
+
+	return record
 end
 
 ---@param match table
----@return table
-function CustomMatchGroupInput._getOpponents(match)
-	-- read opponents and ignore empty ones
-	for opponentIndex = 1, MAX_NUM_OPPONENTS do
-		-- read opponent
-		local opponent = match['opponent' .. opponentIndex]
-		if Logic.isNotEmpty(opponent) then
-			CustomMatchGroupInput.processOpponent(opponent, match.timestamp)
-		end
-		match['opponent' .. opponentIndex] = opponent
-
-		if opponent.type == Opponent.team and Logic.isNotEmpty(opponent.template) then
-			local template = mw.ext.TeamTemplate.raw(opponent.template)
-			if template then
-				MatchGroupInput.readPlayersOfTeam(match, opponentIndex, template.page, {
-					resolveRedirect = true,
-					applyUnderScores = true,
-					maxNumPlayers = MAX_NUM_PLAYERS,
-				})
-			end
+---@param opponents table[]
+---@return table[]
+function CustomMatchGroupInput.extractMaps(match, opponents)
+	local maps = {}
+	for key, map in Table.iter.pairsByPrefix(match, 'map', {requireIndex = true}) do
+		local finishedInput = map.finished --[[@as string?]]
+		local winnerInput = map.winner --[[@as string?]]
+
+		map.extradata = {}
+		map.map, map.extradata.displayname = CustomMatchGroupInput._getMapName(map, match.mapsInfo)
+		map.extradata.mapmode = map.mode
```

Comment:
do we need to Table.extract here?
(so mode gets applied in `getCommonTournamentVars` without the mapmode being used there)